openhtf/core/test_record.py

```diff
@@ -53,8 +53,8 @@ class TestRecord(  # pylint: disable=no-init
         'TestRecord', ['dut_id', 'station_id'],
         {'start_time_millis': None, 'end_time_millis': None,
          'outcome': None, 'outcome_details': list,
-         'code_info': None,
-         'metadata': dict,
+         'code_info': None, 'metadata': dict,
+         'pending_phases': list,
```

Comment:
TestRecord is intended to be an append-only output mechanism, not a place to store transient state.  I'd rather see a station_api mechanism for including a list of all the phase names in the test metadata, and then have the frontend server use that in coordination with phase completion notifications to internally track pending phases.